### PR TITLE
feat: [CI-14878]: Use client certs for mTLS connection

### PR DESCRIPTION
**Pull request by devkimittal**
loading mtls cert and key from /etc/mtls/client.crt, /etc/mtls/client.key for manager client

### CODE REVIEW

File: delegate/http.go [hunk 1]
```diff
@@ -64,68 +64,90 @@ func getClient(endpoint, id, token string, cache *TokenCache, skipverify bool, a
...
-			log.Errorf("could not read directory %s, error: %s", additionalCertsDir, err)
-			c.Client = clientWithRootCAs(skipverify, rootCAs)
-			return c
+			fmt.Println(fmt.Errorf("could not read certificate file (%s), error: %s", path, err.Error()))
+			continue
+		}
+		// Append certs to the global certs
+		ok := rootCAs.AppendCertsFromPEM(rootPem)
+		if !ok {
+			fmt.Println(fmt.Errorf("error adding cert (%s) to pool, please check format of the certs provided", path))
```
Comment (anurag-harness) [inline delegate/http.go:113]: Switch to log.Errorf. PR checks are failing

**Review by anurag-harness — CHANGES_REQUESTED** [review]

**Comment by vistaarjuneja** [comment]
dlite is only used for hosted - do we need to make changes for mtls there as well? @devkimittal @anurag-harness

**Comment by anurag-harness** [comment]
> dlite is only used for hosted - do we need to make changes for mtls there as well? @devkimittal @anurag-harness

We use the same client in k8 engine to send step status from engine to manager. The change is to support mtls in that

**Review by anurag-harness — APPROVED** [review]

File: delegate/http.go [hunk 2]
```diff
@@ -64,68 +64,90 @@ func getClient(endpoint, id, token string, cache *TokenCache, skipverify bool, a
...
-			rootCAs = x509.NewCertPool()
-		}
+	// Load mTLS certificates if available
+	mtlsEnabled, mtlsCerts := loadMTLSCerts(log, "/etc/mtls/client.crt", "/etc/mtls/client.key")
+
+	// Load custom root CAs if additional certificates directory is provided
+	rootCAs := loadRootCAs(log, additionalCertsDir)
+
+	// Only create HTTP client if needed (mTLS, additional certs, or skipverify)
+	if skipverify || rootCAs != nil || mtlsEnabled {
```
Comment (hemanthmantri) [inline delegate/http.go:75]: What is `skipVerify`?  Shouldn't this be  `if !skipVerify` ?

**Pull request merged**